routers/api/v1/repo/commits.go

```diff
@@ -55,10 +67,19 @@ func GetSingleCommit(ctx *context.APIContext) {
 		ctx.Error(http.StatusUnprocessableEntity, "no valid ref or sha", fmt.Sprintf("no valid ref or sha: %s", sha))
 		return
 	}
-	getCommit(ctx, sha)
+
+	stat := ctx.FormString("stat") == "" || ctx.FormBool("stat")
+	files := ctx.FormString("files") == "" || ctx.FormBool("files")
+	verification := ctx.FormString("verification") == "" || ctx.FormBool("verification")
+
+	getCommit(ctx, sha, convert.ToCommitOptions{
+		Stat:         stat,
+		Files:        files,
+		Verification: verification,
+	})
```

Comment:
Instead of duplicating this code, what about writing a helper function `ParseCommitOptions(ctx)` or something like that that returns the created commit options?

Reply:
Good question…
Perhaps `services/convert/git_commit.go`?